src/supported-operations/running-monitoring-operations.md

```diff
@@ -0,0 +1,94 @@
+  - Has the process to be detached from the agent daemon?
+  - Has an operation to be aborted on agent restart?
+
+## Proposed design
+* The operations are defined by the configuration files in [`/etc/tedge/operations`](./principles.md).
+* The cloud mappers trigger the operations.
+  * This is done on reception of operation requests from the cloud:
+    each operation configuration defines the topic to listen, and the message-pattern to match.
+  * The responsibility of the mapper is only to translate cloud specific messages into thin-edge requests.
+    When an operation request is received, the mapper combines the request and the operation configuration
+    into execution information, and delegates the execution to the agent.
+* The agent is the process that runs and monitors the operation commands.
+  * The agent listen for operation request on the `tedge/operations/start` topic.
+  * Each request provides the information to execute and monitor the operation:
+     * a request unique identifier,
+     * the operation name,
+     * the command path,
+     * an array of command arguments (at least the message that triggered the request),  
+     * the user running the command.
+  * ```json
+    { "id": "abcdef",
+      "operation": "c8y_LogfileRequest",
+      "command": "/usr/bin/c8y_logfile",
+      "args": ["522,DeviceSerial,logfileA,2013-06-22T17:03:14.000+02:00,2013-06-22T18:03:14.000+02:00,ERROR,1000"],
+      "user": "tedge"
+    }
+    ```
+* The agent schedules the operations reporting the progress on the `tedge/operations/status` topic.
+  * ```json
+    { "id": "abcdef",
+      "operation": "c8y_LogfileRequest",
+      "status": "scheduled"
+    }
+    ```
+  * `"status": "schedule"`: on reception of a well-formatted operation request.
+  * `"status": "executing"`: just before executing the operation.
+  * `"status": "successful"`: when the operation returns with a zero exit status.
+  * `"status": "failed"`: when the operation returns with a non-zero status.
+  * NOTE: One might also consider `cancelled` and `aborted` operations to deal with user cancellation
+    and operations killed by a signal or a power failure.
+* The mapper might subscribe to the `tedge/operations/status` topic
+  and translate the operation status into cloud specific messages.
+  * NOTE: There is open question here.
+    * Does the c8y mapper translate the `tedge/operations/status` messages into equivalent smartRest2 messages?
+    * Or does the c8y operation publish their progress directly to `c8y/s/us`?
+    * The former would be simpler for the operator implementor. 
+      However, most operations will in any case have to dialog over `c8y/s/us` and `c8y/s/ds`
+      to implement the specificities of the operation protocol.
+* The execution of a command is logged in file name after the operation name and the request id.
+  * `/var/log/tedge/agent/{operation-name}-{operation-request-id}-{starting-datetime}.log`
```

Comment:
These logs can be requested from the cloud to view as well right?

Reply:
Yes. The upload log feature will have to be extended.